agent-shell.el

```diff
@@ -389,6 +455,88 @@ Returns an empty string if no icon should be displayed."
   (interactive)
   (message "agent-shell v%s" agent-shell--version))
 
+;;; Extension API
+
+(defun agent-shell-get-state (&optional buffer)
+  "Get agent shell state from BUFFER (defaults to current buffer).
+Returns nil if buffer is not an agent-shell buffer.
+Note: The returned state should be treated as read-only by extensions."
+  (with-current-buffer (or buffer (current-buffer))
+    (when (and (boundp 'agent-shell--state)
+               (derived-mode-p 'agent-shell-mode))
+      agent-shell--state)))
+
+(defun agent-shell-get-client (&optional buffer)
+  "Get ACP client from BUFFER's agent shell state.
+BUFFER defaults to current buffer.
+Returns nil if buffer is not an agent-shell buffer."
+  (map-elt (agent-shell-get-state buffer) :client))
+
+(defun agent-shell-tool-call-get (tool-call-id &optional buffer)
+  "Get tool call data for TOOL-CALL-ID from BUFFER's agent shell state.
+BUFFER defaults to current buffer.
+Returns the tool call alist or nil if not found."
+  (when-let ((state (agent-shell-get-state buffer)))
+    (map-nested-elt state (list :tool-calls tool-call-id))))
+
+(defun agent-shell-tool-call-put (tool-call-id data &optional buffer)
+  "Store tool call DATA for TOOL-CALL-ID in BUFFER's agent shell state.
+BUFFER defaults to current buffer.
+DATA should be an alist that will be merged with existing data.
+Returns non-nil on success."
+  (when-let ((state (agent-shell-get-state buffer)))
+    (agent-shell--save-tool-call state tool-call-id data)
+    t))
+
+(defun agent-shell-resolve-path (path)
```

Comment:
(cl-defun agent-shell-resolve (&key path)

(unless path
  ...)

Client code will be fairly explict:

`(agent-shell-resolve :path "some/path/here")`